rest/processGlobalQueue.ts

```diff
@@ -118,18 +118,33 @@ export async function processGlobalQueue(rest: RestManager) {
 
         // If NOT rate limited remove from queue
         if (response.status !== 429) {
+          let json = undefined;
           if (response.type) {
-            console.log(JSON.stringify(await response.json()));
+            json = await response.json();
+            console.log(JSON.stringify(json));
           }
 
-          request.request.reject(new Error(`[${response.status}] ${error}`));
+          request.request.reject({
+            ok: false,
+            status: response.status,
+            error: {
+              raw: new Error(`[${response.status}] ${error}`),
+              name: error,
+            },
```

Comment:
Completely remove `new Error()` and make the `runMethod` function construct it.

```suggestion
            error,
```